### DISCUSSION TITLE
About COM_receiver_noise_filter

### DISCUSSION THREAD

**Discussion by YasuoKawahara** [scikit-rf/scikit-rf#1098] [Q&A] [answered]
Hello,
There is "receiver filter in COM defined by eq 93A-20" in comment of COM_receiver_noise_filter.
I want more information. What is COM? What is eq 93A-20?
I would be very grateful if you could give me some useful URLs etc.

Kawahara

Edit: converted from issue to discussion.

**Comment by mhuser** [comment, marked as answer]
Hello @YasuoKawahara 

```Python
    def COM_receiver_noise_filter(self, f,fr):
        """
        receiver filter in COM defined by eq 93A-20
        """
        fdfr = f / fr
        # eq 93A-20
        return 1 / (1 - 3.414214 * fdfr**2 + fdfr**4 + 1j*2.613126*(fdfr - fdfr**3))
```

[`COM_receiver_noise_filter`](https://github.com/scikit-rf/scikit-rf/blob/3ba8bf84aa07f9dab0ab70992c70dbc178f7929e/skrf/calibration/deembedding.py#L1114C1-L1117C12) is a feature of the [IEE370](https://standards.ieee.org/ieee/370/6165/) implementation.

The `scikit-rf` implementation is a clone of the [Matlab implementation](https://opensource.ieee.org/elec-char/ieee-370/-/blob/master/TG1/IEEEP3702xThru.m) that is an asset of IEEE370.
It was compared against the Matlab results for [some cases](https://scikit-rf.readthedocs.io/en/latest/examples/networktheory/IEEEP370%20Deembedding.html) and is believed to present a good agreement.

You can find more inside Ellison J, Smith SB, Agili S., "Using a 2x-thru standard to achieve
        accurate de-embedding of measurements", Microwave Optical Technology
        Letter, 2020, https://doi.org/10.1002/mop.32098

But the true answers you are looking for are in [802.3-2022 - IEEE Standard for Ethernet](https://ieeexplore.ieee.org/document/9844436) annex 93A, that defines various filters equations.
![image](https://github.com/scikit-rf/scikit-rf/assets/8080934/c21dcda7-3d17-4ca2-8f88-98f9ee241f0f)

**Reply by denzchoe** [reply]
Just to add some side info, not related to scikit-rf: COM is actually defined as Channel Operating Margin.
You can see how it is measured here: https://dl.cdn-anritsu.com/en-us/test-measurement/files/Technical-Notes/White-Paper/11410-00989A.pdf

And the actual MATLAB code used for COM in the Current and Upcoming IEEE802.3dj Standards are presently maintained here: 
https://www.ieee802.org/3/dj/public/adhoc/COM/index.html
